DataFormats/EcalDigi/interface/EcalDigiPhase2SoA.h

```diff
@@ -0,0 +1,21 @@
+#ifndef DataFormats_EcalDigi_EcalDigiPhase2SoA_h
+#define DataFormats_EcalDigi_EcalDigiPhase2SoA_h
+
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+
+namespace ecal {
+
+  // due to a ROOT limitation the std::array needs to be wrapped in a struct
+  // https://github.com/root-project/root/issues/12007
+  using DataArrayPhase2Struct = StdArrayStruct<uint16_t, ecalPh2::sampleSize>;
+
+  GENERATE_SOA_LAYOUT(EcalDigiPhase2SoALayout,
+                      SOA_COLUMN(uint32_t, id),
+                      SOA_COLUMN(DataArrayPhase2Struct, data),
```

Comment:
The `sampleSize` seems to be 16 (phase1) or 10 (phase2) So this column has 16-or-10 2-byte sub-elements per SoA-element (i.e. 32-or-20 bytes).

How is this data structure accessed by the consuming code? Does the consuming code use a GPU-thread per element of this SoA? If yes, then I'm wondering if it would be more performant to split this column into `sampleSize` arrays with Eigen vectors? In principle that should yield memory access pattern that is more coalesced than this one.

(as far as I'm concerned this optimization can be done later, it would anyway be better to study the impact on the  performance rather than to rely on anything I wrote above on the top of my head) 

(this pattern repeats in other SoAs as well, I'm not commenting on them in order to have one place to discuss about it)

Reply:
It should be the other way around, in fact. 10 elements for Phase 1 and 16 for Phase 2.
At the moment one thread will use all 10 or 16 elements to calculate the weighted sum in the Phase 2 case.
I am not sure how the fitting in the multifit algorithm works but I think if also uses all samples to perform the fit in one thread.